teammapper-frontend/src/app/core/services/mmp/mmp.service.ts

```diff
@@ -563,4 +494,101 @@ export class MmpService implements OnDestroy {
       fontIncrement: defaultSettings.mapOptions.fontIncrement,
     };
   }
+
+  /**
+   * Export map to image (png, jpg, svg)
+   */
+  private async exportToImage(format: string, name: string) {
+    const image = await this.exportAsImage(format);
+
+    if (image === EMPTY_IMAGE_DATA) {
+      const exportImageFailureMessage = await this.utilsService.translate(
+        'TOASTS.ERRORS.EXPORT_IMAGE_ERROR'
+      );
+      this.toastrService.error(exportImageFailureMessage);
+      return { success: false };
+    } else {
+      UtilsService.downloadFile(
+        `${name}.${format === 'jpeg' ? 'jpg' : format}`,
+        image
+      );
+
+      return { success: true, size: image.length / 1024 };
+    }
+  }
+
+  /**
+   * Export map to PDF
+   */
+  private async exportToPDF(format: string, name: string) {
```

Comment:
fyi: As far as I know its good practice to always define a typescript return type to functions. Yes, the return value can be inferred from the return value. But to make sure the return matches the writers specific , it's good to add it.

fyi2: pr ist still in draft mode

Reply:
Return type can be done. It's still in draft mode because I want to implement the node boundary in this PR as well, but it's _complicated_.